SplitIt/app/src/main/java/project/cs495/splitit/ReceiptViewActivity.java

```diff
@@ -408,15 +407,18 @@ private void createUserReceipt() {
         mDatabaseReference.child(getString(R.string.items)).orderByChild(getString(R.string.receipt_ids_path)+receiptId).equalTo(true).addListenerForSingleValueEvent(new ValueEventListener() {
             @Override
             public void onDataChange(DataSnapshot dataSnapshot) {
-                float totalPrice = 0;
+                float subtotal = 0;
                 for (DataSnapshot itemSnapshot : dataSnapshot.getChildren()) {
                     Item item = itemSnapshot.getValue(Item.class);
                     if (item.getAssignedUser() != null && item.getAssignedUser().equals(userId)) {
                         userReceipt.addItem(item.getItemId());
-                        totalPrice += item.getPrice();
+                        subtotal += item.getPrice();
                     }
                 }
-                userReceipt.setPrice(totalPrice);
+                userReceipt.setSubtotal(subtotal);
+                float percentOfSubtotal = subtotal / receipt.getSubtotal();
+                userReceipt.setTax(receipt.getTax() * percentOfSubtotal);
+                userReceipt.setPrice(subtotal + userReceipt.getTax());
                 userReceipt.commitToDB(mDatabaseReference);
```

Comment:
Get this error when trying to generate a user receipt: "com.google.firebase.database.DatabaseException: Invalid value: Value cannot be NaN, Inf or -Inf. 
at project.cs495.splitit.models.UserReceipt.commitToDB(UserReceipt.java:21)
at project.cs495.splitit.ReceiptViewActivity$11.onDataChange(ReceiptViewActivity.java:422)"

Strangely it doesn't happen when I create a user receipt from the target receipt you shared with me, but it does with receipts I already had scanned.

Reply:
I'm guessing it has to do with those prior receipts not having the tax info stored on them. I'll look into it